src/client/src/StreamState.c

```diff
@@ -705,14 +705,34 @@ STATUS executeStoppedStreamState(UINT64 customData, UINT64 time)
     ENTERS();
     UNUSED_PARAM(time);
     STATUS retStatus = STATUS_SUCCESS;
+    UINT64 duration, viewByteSize;
     PKinesisVideoStream pKinesisVideoStream = STREAM_FROM_CUSTOM_DATA(customData);
 
     CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
 
     // Also store the connection stopping result
     pKinesisVideoStream->connectionDroppedResult = pKinesisVideoStream->base.result;
 
-    // Step the client state machine first
+    // Check if we want to prime the state machine based on whether we have any more content to send
+    // currently and if the error is a timeout.
+    if (SERVICE_CALL_RESULT_IS_A_TIMEOUT(pKinesisVideoStream->connectionDroppedResult)) {
+        CHK_STATUS(getAvailableViewSize(pKinesisVideoStream, &duration, &viewByteSize));
+        if (viewByteSize == 0) {
+            // Next time putFrame is called we will self-prime.
+            // The idea is to drop frames till new key frame which will
+            // become a stream start.
+            pKinesisVideoStream->resetGeneratorOnKeyFrame = TRUE;
```

Comment:
is this gonna cause two mkv headers to be generated? we also generate header in streamStartFixupOnReconnect.

Reply:
A single header. The changes in the latest test hits this case to validate.